tests/unit/s2n_tls13_cert_verify_test.c

```diff
@@ -313,6 +313,8 @@ int run_tests(const struct s2n_tls13_cert_verify_test *test_case, s2n_mode verif
         EXPECT_SUCCESS(s2n_hash_init(&verifying_conn->handshake.hashes->sha256, S2N_HASH_SHA256));
         EXPECT_SUCCESS(s2n_hash_update(&verifying_conn->handshake.hashes->sha256, hello, strlen((char *) hello)));
 
+        test_scheme.sig_alg = S2N_SIGNATURE_ECDSA;
```

Comment:
I understand that this gets the test to pass, but I don't really understand why it does. What happens when you print out the verifying_conn->handshake_params.client_cert_sig_scheme and server_cert_sig_scheme for this test case?

I'm not sure I fully understand why the original test was failing.

Reply:
Oh I think I know why the original test was failing. 
This test gets run in a for loop:
`int main(int argc, char **argv)
{
    BEGIN_TEST();

    EXPECT_SUCCESS(s2n_enable_tls13_in_test());

    for (size_t i = 0; i < sizeof(test_cases) / sizeof(struct s2n_tls13_cert_verify_test); i++) {
        /* Run all tests for server sending and client receiving/verifying cert_verify message */
        run_tests(&test_cases[i], S2N_CLIENT);

        /* Run all tests for client sending and server receiving/verifying cert_verify message */
        run_tests(&test_cases[i], S2N_SERVER);
    }

    END_TEST();
}`

The first iteration uses s2n_ecdsa_sha256 as signature scheme for the test and the second iteration uses s2n_rsa_pss_pss_sha256 as the signature scheme. The first iteration passes, but the second iteration fails because we're setting test_scheme.sig_alg = S2N_SIGNATURE_ECDSA; and this causes mismatch with RSA_PSS_PSS. If we set them dynamically to the correct sig_alg the test should pass with correct failure.